src/main/java/sirius/kernel/commons/Trie.java

```diff
@@ -329,4 +331,65 @@ public void put(@Nonnull CharSequence key, @Nonnull V value) {
         }
         iter.setValue(value);
     }
+
+    /**
+     * Retrieves all keys that are stored in this {@link Trie}.
+     *
+     * @return a {@link Set} of all keys that are stored in this {@link Trie}
+     */
+    public Set<String> keySet() {
+        return getAllKeysBeginningWith("");
+    }
+
+    /**
+     * Retrieves the number of keys that are stored in this {@link Trie}.
+     *
+     * @return the size of this {@link Trie}'s {@link #keySet() key set}
+     */
+    public int size() {
+        return keySet().size();
+    }
+
+    /**
+     * Performs a prefix search within this {@link Trie}'s {@link #keySet() key set}
+     *
+     * @param prefix to search for
+     * @return all keys that are beginning with the given <tt>prefix</tt> (may include <tt>prefix</tt> itself)
+     */
+    public Set<String> getAllKeysBeginningWith(String prefix) {
+        ContainmentIterator<V> iter = iterator();
+        for (int i = 0; i < prefix.length(); i++) {
+            if (!iter.doContinue(prefix.charAt(i))) {
+                return Collections.emptySet();
+            }
+        }
+        return getAllKeysBeginningWith(prefix, iter);
+    }
+
+    private Set<String> getAllKeysBeginningWith(String prefix, ContainmentIterator<V> iter) {
+        if (iter.getPossibilities().isEmpty()) {
+            if (iter.getValue() != null) {
+                return Sets.newHashSet(prefix);
+            } else {
+                return Collections.emptySet();
+            }
+        }
+
+        Set<String> result = new HashSet<>();
+        if (iter.getValue() != null) {
+            result.add(prefix);
+        }
+        for (char possibility : iter.getPossibilities()) {
+            iter.doContinue(possibility);
+            result.addAll(getAllKeysBeginningWith(prefix + possibility, iter));
+            iter.goBack();
+        }
+
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return Strings.join(new TreeSet<>(keySet()), ", ");
```

Comment:
isn't that a bad idea, considering that Tries easily collect 10.000s of entries...?